Assets/Scripts/UI/PepemonFactoryCardCache.cs

```diff
@@ -76,16 +63,63 @@ public static async Task PreloadAllImages(ulong parallelBatchSize, Action<ulong>
         }
     }
 
-
-    private static async Task PreloadTokenMetadata(ulong tokenId, Action<ulong> cardLoadedCallback = null)
+    private static async UniTask PreloadTokenMetadata(ulong minId, ulong maxId, Action<ulong> cardLoadedCallback = null)
     {
-        PepemonFactory.CardMetadata? metadata = await PepemonFactory.GetCardMetadata(tokenId);
-        if (metadata != null)
+        var supportCardsToFetch = new List<ulong>();
+        var battleCardStats = await PepemonFactory.BatchGetBattleCardStats(minId, maxId);
+        for (var i = 0; i < battleCardStats.Count; i++)
+        {
+            var tokenId = (ulong)i + minId;
+
+            // when we accidentally tried to fetch a battlecard but it was a support card
+            if (battleCardStats[i].Hp == 0)
+            {
+                supportCardsToFetch.Add(tokenId);
+            }
+            else
+            {
+                cardMetadata[tokenId] = new CardMetadata
+                {
+                    description = battleCardStats[i].Description,
+                    image = battleCardStats[i].IpfsAddr,
+                    isSupportCard = false,
+                    name = battleCardStats[i].Name
+                };
+                cardLoadedCallback?.Invoke(tokenId);
+                Debug.Log($"Loaded token metadata for id: {tokenId}");
+            }
+        }
+        if (supportCardsToFetch.Count == 0)
+        {
+            return;
+        }
+
+        List<SupportCardStats> supportCardStats;
+        try
+        {
+            supportCardStats = await PepemonFactory.BatchGetSupportCardStats(
+                supportCardsToFetch[0], supportCardsToFetch[supportCardsToFetch.Count - 1]);
+        }
+        catch (Exception ex)
+        {
+            Debug.LogException(ex);
+            return;
+        }
+
+        for (var i = 0; i < supportCardStats.Count; i++)
         {
-            cardMetadata[tokenId] = (PepemonFactory.CardMetadata)metadata;
+            var tokenId = (ulong)i + minId;
+
+            cardMetadata[tokenId] = new CardMetadata
+            {
+                description = supportCardStats[i].Description,
+                image = supportCardStats[i].IpfsAddr,
+                isSupportCard = true,
+                name = supportCardStats[i].Name
+            };
+            cardLoadedCallback?.Invoke(tokenId);
+            Debug.Log($"Loaded token metadata for id: {tokenId}");
```

Comment:
Won't the `minId` and `maxId` passed to `BatchGetSupportCardStats` also include battle cards at some point? I don't see any filtering for that here